### PR TITLE
Run cargo build and test built binary to ensure it works

### PR DESCRIPTION
**Pull request by tonowak**
Closes https://github.com/obi1kenobi/cargo-semver-checks/issues/235

### CODE REVIEW

**Review by epage — COMMENTED** [review]
imo if you have this complex of a step, it should probably broken out.

What I'd recommend doing is writing this as a test. Cargo builds binaries before running tests, so you can write end-to-end cli tests in Rust. I used to write mine with assert_cmd/assert_fs but have switched to snapbox

**Comment by tonowak** [comment]
Thanks for the information! I wrote the test using `assert_cmd`.

File: Cargo.toml [hunk 1]
```diff
@@ -37,3 +37,4 @@ bugreport = "0.5.0"
 itertools = "0.10.5"
 cargo_toml = "0.13.0"
 toml = "0.5.9"
+assert_cmd = "2.0"
```
Comment (epage) [inline Cargo.toml:40]: This should be added as a `dev-dependency`

File: src/main.rs [hunk 2]
```diff
@@ -282,8 +282,43 @@ struct CheckRelease {
     verbosity: clap_verbosity_flag::Verbosity<clap_verbosity_flag::InfoLevel>,
 }
 
-#[test]
-fn verify_cli() {
-    use clap::CommandFactory;
-    Cargo::command().debug_assert()
+#[cfg(test)]
+mod tests {
```
Comment (epage) [inline src/main.rs:286]: > thread 'tests::verify_binary_contains_lints' panicked at 'called `Result::unwrap()` on an `Err` value: CargoError { cause: Some(NotFoundError { path: "/home/runner/work/cargo-semver-checks/cargo-semver-checks/target/debug/cargo-semver-checks" }) }', src/main.rs:298:69

I think the guarantee that the binary is built first is only available for "integration" / "black box" tests (in the `tests/` directory)

File: src/main.rs [hunk 3]
```diff
@@ -282,43 +282,8 @@ struct CheckRelease {
...
-            "function_const_removed",
-            "function_unsafe_added",
-        ] {
-            assert_on_crate_pair(crate_pair).failure();
-        }
-    }
+#[test]
+fn verify_cli() {
+    use clap::CommandFactory;
+    Cargo::command().debug_assert()
```
Comment (obi1kenobi) [inline src/main.rs:288]: Did you mean to change this and pull it out of `mod tests`?

Comment (tonowak) [reply src/main.rs:288]: I decided to keep `src/main.rs` how it was originally (by simply reverting the `src/main.rs` after my changes to the state before my changes). Should I nevertheless move it to `mod tests`?

**Review by obi1kenobi — APPROVED** [review]

File: src/main.rs (hunk 3 above)
Comment (obi1kenobi) [reply src/main.rs:288]: Sorry, I must have been looking at a stale version of the PR.

**Pull request merged**